wpilibj/src/main/java/edu/wpi/first/wpilibj/ResistanceCalculator.java

```diff
@@ -0,0 +1,134 @@
+   *
+   * @param bufferSize The maximum number of points to take the linear regression over.
+   * @param krSquaredThreshold The minimum R^2 value considered significant enough to return the
+   *     regression slope instead of NaN.
+   */
+  public ResistanceCalculator(int bufferSize, double krSquaredThreshold) {
+    m_currentBuffer = new CircularBuffer(bufferSize);
+    m_voltageBuffer = new CircularBuffer(bufferSize);
+    this.m_rSquaredThreshold = krSquaredThreshold;
+    this.m_bufferSize = bufferSize;
+  }
+
+  /**
+   * Create a {@code ResistanceCalculator} to find the resistance of a channel using a running
+   * linear regression over a window. Must be updated with current and voltage periodically using
+   * the {@link ResistanceCalculator#calculate(double, double) calculate} method.
+   *
+   * <p>Uses a buffer size of {@link ResistanceCalculator#kDefaultBufferSize} and an r^2 threshold
+   * of {@link ResistanceCalculator#kDefaultRSquaredThreshold}.
+   */
+  public ResistanceCalculator() {
+    m_currentBuffer = new CircularBuffer(kDefaultBufferSize);
+    m_voltageBuffer = new CircularBuffer(kDefaultBufferSize);
+    this.m_rSquaredThreshold = kDefaultRSquaredThreshold;
+    this.m_bufferSize = kDefaultBufferSize;
+  }
+
+  /**
+   * Update the buffers with new (current, voltage) points, and remove old points if necessary.
+   *
+   * @param current The current current
+   * @param voltage The current voltage
+   * @return The current resistance, in ohms
+   */
+  public double calculate(double current, double voltage) {
+    // Update buffers only if drawing current
+    if (current != 0) {
+      if (m_numPoints >= m_bufferSize) {
+        // Pop the last point and remove it from the sums
+        double backCurrent = m_currentBuffer.removeLast();
+        double backVoltage = m_voltageBuffer.removeLast();
+        m_currentSum -= backCurrent;
+        m_voltageSum -= backVoltage;
+        m_currentSquaredSum -= backCurrent * backCurrent;
+        m_voltageSquaredSum -= backVoltage * backVoltage;
+        m_prodSum -= backCurrent * backVoltage;
+      } else {
+        m_numPoints++;
+      }
+      m_currentBuffer.addFirst(current);
+      m_voltageBuffer.addFirst(voltage);
+      m_currentSum += current;
+      m_voltageSum += voltage;
+      m_currentSquaredSum += current * current;
+      m_voltageSquaredSum += voltage * voltage;
+      m_prodSum += current * voltage;
+    }
+
+    // Recalculate resistance
+    if (m_numPoints < 2) {
+      return Double.NaN;
+    }
+
+    double currentVariance =
+        (m_currentSquaredSum / m_numPoints) - Math.pow(m_currentSum / m_numPoints, 2);
+    double voltageVariance =
+        (m_voltageSquaredSum / m_numPoints) - Math.pow(m_voltageSum / m_numPoints, 2);
+    double covariance = (m_prodSum - m_currentSum * m_voltageSum / m_numPoints) / (m_numPoints - 1);
+    double krSquared = covariance * covariance / (currentVariance * voltageVariance);
```

Comment:
```suggestion
    double rSquared = covariance * covariance / (currentVariance * voltageVariance);
```
The k prefix is reserved for constants.

Reply:
The Java linter was unhappy before about `rSquared` starting with a single lowercase letter, so I figured I'd put in an extra letter and forgot to make it just `rSquared` in C++.

Reply:
Suppress the linter warning. rSquared is the proper mathematical name for it.